test/controllers/UpfrontPaymentControllerSpec.scala

```diff
@@ -21,22 +21,66 @@ import play.api.mvc.Result
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import testsupport.ItSpec
+import testsupport.TdRequest.FakeRequestOps
+import testsupport.stubs.{AuthStub, EssttpBackend}
+import testsupport.testdata.TdAll
+import uk.gov.hmrc.http.SessionKeys
 
 import scala.concurrent.Future
 
 class UpfrontPaymentControllerSpec extends ItSpec {
 
   private val controller: UpfrontPaymentController = app.injector.instanceOf[UpfrontPaymentController]
 
-  "GET /upfrontpayment or whatever - Pawel, shall we add tests like these or just stick with int?" - {
-    "return 200 and the upfront payment page" ignore {
-      val fakeRequest = FakeRequest("GET", "/upfront-payment")
-      val result: Future[Result] = controller.upfrontPayment(fakeRequest)
+  "GET /can-you-make-an-upfront-payment" - {
+    "return 200 and the can you make an upfront payment page" in {
```

Comment:
what do you think about also testing to see if nothing is selected on the form if there isn't an existing answer in session and also testing to see that the correct thing is selected if there is an existing answer in session?

Other possible things to test are also the back link location or the href of the submit button.